README.md

```diff
@@ -1,133 +1,416 @@
+- For `key` option, user needs to specify the primary key column of SQLite table corresponding with the `key` option. If not, wrong result may occur when update or delete.
+- When `Sum` of data in table is out of range, `sqlite_fdw` will display `Infinity` value. It is different from PostgreSQL FDW, which will display `ERROR: value out of range: overflow` error.
+- For `numeric` data type, `sqlite_fdw` use `sqlite3_column_double` to get value, while SQLite shell uses `sqlite3_column_text` to get value. Those 2 APIs may return different numeric value. Therefore, for `numeric` data type, the value returned from `sqlite_fdw` may different from the value returned from SQLite shell.
+- `sqlite_fdw` can return implementation-dependent order for column if the column is not specified in `ORDER BY` clause.
+- When the column type is `varchar array`, if the string is shorter than the declared length, values of type character will be space-padded; values of type `character varying` will simply store the shorter string.
+
+Also see [Limitations](#limitations)
+
+Supported platforms
+-------------------
+
+`sqlite_fdw` was developed on Linux and should run on any
+reasonably POSIX-compliant system.
+
+Installation
+------------
+
+For some Linux distributives deb and rpm packages are avalillable.
+
+### Source installation
+
+Prerequisites:
+* `libsqlite3-dev`, especially `sqlite.h`
+* `gcc`
+* `make`
+
+#### 1. Install SQLite library
+
+For Debian or Ubuntu:
+`apt-get install libsqlite3-dev`
+
+You can also [download SQLite source code][1] and [build SQLite][2].
+
+#### 2. Build and install sqlite_fdw
+
+Add a directory of `pg_config` to PATH and build and install `sqlite_fdw`.
+
+```sh
 make USE_PGXS=1
 make install USE_PGXS=1
-</pre>
+```
 
-If you want to build sqlite_fdw in a source tree of PostgreSQL, use
-<pre>
+If you want to build `sqlite_fdw` in a source tree of PostgreSQL, use
+```sh
 make
 make install
-</pre>
+```
+
+Usage
+-----
+
+## CREATE SERVER options
+
+`sqlite_fdw` accepts the following options via the `CREATE SERVER` command:
+
+- **database** as *string*, **required**
+
+  SQLite database path.
+
+- **truncatable** as *boolean*, optional
+
+  Allows foreign tables to be truncated using the `TRUNCATE` command.
+  
+- **keep_connections** as *boolean*, optional
+  
+  Allows to keep connections to SQLite while there is no SQL operations between PostgreSQL and SQLite.
+  
+- **batch_size** as *integer*, optional
+
+  Specifies the number of rows which should be inserted in a single `INSERT` operation. This setting can be overridden for individual tables.
+
+
+## CREATE USER MAPPING options
+
+`sqlite_fdw` no need any `CREATE USER MAPPING` command.
+
+Just ensure `postgres` OS user have permissions for reading or writing on SQLite database file.
+
+
+## CREATE FOREIGN TABLE options
+
+`sqlite_fdw` accepts the following table-level options via the
+`CREATE FOREIGN TABLE` command:
+
+- **table** as *string*, **required**
```

Comment:
I think this option is not required. It should be used only when SQLite table name is different from foreign table name.
If user does not provide it, it uses the foreign table name by default.
Therefore, it should be optional.
Our README also describes it as required. It seems this is a mistake from our side. Please help us to update it.

Reply:
Thanks, @son-phamngoc ! Corrected.

Reply:
Please fix the typo: optioanl